src/Common/tests/System/AssertExtensions.cs

```diff
@@ -14,5 +15,17 @@ public static void Throws<T>(Action action, string message)
         {
             Assert.Equal(Assert.Throws<T>(action).Message, message);
         }
+
+        public static void Throws<T>(string netCoreParamName, string netFxParamName, Action action)
+            where T : ArgumentException
+        {
+            T exception = Assert.Throws<T>(action);
+
+            string expectedParamName =
+                RuntimeInformation.FrameworkDescription.StartsWith(".NET Framework") ?
+                netFxParamName : netCoreParamName;
```

Comment:
In master we are only running tests against the latest. The only test runs we do against the older versions are in the release branches, so as long as this change doesn't flow there then we should be good.